### ISSUE TITLE
Please add Wayland support

### ISSUE THREAD

**Issue by ilan0098** [catsout/wallpaper-engine-kde-plugin#302]
The plugin works just fine for me on X11 , but I get black screen on Wayland , can it be fixed by anyone?

**Comment by Jelgnum** [comment]
it does work on Wayland, currently using it myself

**Comment by barfin** [comment]
it already works just fine
![image](https://github.com/catsout/wallpaper-engine-kde-plugin/assets/37030489/a1782b67-2a66-4b4b-a494-441161db18be)

**Comment by ilan0098** [comment]
Ohh , I tried using it myself and it won't work for some reason, shows black screen

**Comment by barfin** [comment]
> Ohh , I tried using it myself and it won't work for some reason, shows black screen

Try a different (and preferably simple) wallpaper, sometimes when a wallpaper isn't supported it shows black screen

**Comment by ilan0098** [comment]
> 

Yeah , I just found out about it , apperntly it does work on Wayland but sadly it doesn't work for some wallpapers

**Comment by CrazyVito11** [comment]
@ilan0098 Maybe a bit late, but check your system logs.
I also had a black screen, but that was because the video was in a format I didn't have the codec for.
![image](https://github.com/catsout/wallpaper-engine-kde-plugin/assets/31567466/760d7c3c-b47e-4c21-a588-dd9600b7677f)

Once I installed `gst-plugins-good`, it worked just fine.

**Comment by pearmypie** [comment]
Is anyone else experiencing mouse stutters on Wayland? It's as if the FPS of the wallpaper affects the mouse pointer. Even at 60FPS it's not smooth.

**Comment by CrazyVito11** [comment]
@pearmypie I've been running it since I send that message without any issues, I daily drive Wayland.

**Comment by pearmypie** [comment]
@CrazyVito11 I'm on KDE 5.27.5, using the 5.4 version of the plugin lib. If other people running the same versions don't have any problems, I guess I should be looking into driver issues? I'm on Vega 8 AMD integrated graphics.

**Comment by CrazyVito11** [comment]
I'm on KDE 5.27.10, running version 0.5.5 of the plugin and plugin lib, 7900XTX as my GPU.

Try updating to see if that fixes it, otherwise you might have to dig a bit deeper to find the root issue.

**Comment by Martmists-GH** [comment]
Having a similar issue on Wayland; Wallpapers run at ~5-10fps on a display that supports 144fps
